app/services/testing_ground/tree_sampler.rb

```diff
@@ -0,0 +1,31 @@
+class TestingGround
+  class TreeSampler
+    def initialize(networks)
+      @networks = networks
+    end
+
+    def sample(resolution)
+      Hash[@networks.each_pair.map do |carrier, graph|
+        graph.nodes.each do |node|
+          node.set(:load, downsample(node.load, resolution))
+        end
+
+        [carrier, GraphToTree.convert(graph)]
+      end]
+    end
+
+    private
+
+    def downsample(node_load, resolution)
+      if resolution == :low
+        node_load.each_slice(96).map do |loads|
```

Comment:
Instead of hard-coding 96, it would be better to determine this number based on the length of the load curve.

```ruby
class TreeSampler
  RESOLUTION_LENGTH_LOW = 365

  def downsample(*)
    if resolution == :low
      node_load.each_slice((node_load.length / RESOLUTION_LENGTH_LOW).floor)
      # ...
    end
  end
end
```

The reason for this being that the load curve length may not always be 35040; I have an LES locally which uses 8760 (one time-step per hour, instead of one per 15 minutes). This results in the low-res curves containing fewer values than expected.

```ruby
([0.0] * 35040).each_slice(96).map(&:sum) # => 365 values
([0.0] * 8760).each_slice(96).map(&:sum)  # => 96 values
```